src/message/messagesReducer.js

```diff
@@ -137,8 +139,49 @@ export default (
       return state.deleteAll(action.messageIds);
 
     case EVENT_UPDATE_MESSAGE: {
-      // TODO(#3408): Handle these.
-      return state;
+      const { event, move } = action;
+      let result = state;
+
+      if (move) {
+        const update: { subject: string, stream_id?: number, display_recipient?: string } = {
+          subject: move.new_topic,
+        };
+        if (move.new_stream_id !== move.orig_stream_id) {
+          update.stream_id = move.new_stream_id;
+          const newStream = getStreamsById(globalState).get(move.new_stream_id);
+          // It's normal for newStream to potentially be missing here: it
+          // happens when the move was to a stream our user can't see.
+          // TODO(i18n): Not sure this "unknown" ever reaches the UI, but
+          //   it'd be nice to translate it in case it can.
```

Comment:
> I'm not sure if this is what you're suggesting, but is there a precedent for keeping a translated UI string in Redux? I'm not sure how feasible this is in general. To keep the reducers pure, we'd need

Yeah, this sort of reason is why I didn't attempt to actually do this :wink: 

If we had some reasonably simple way to get a translation with the current choice of language, I think because this is already an edge case it could be an acceptable hack here to use that and have the name not get updated if the language choice changes.


> Under the "normalized data" strategy

Yeah, so one consequence of that strategy would be that this lookup would happen in UI code when we're about to present the stream name to the user, instead of here in this reducer. That would make it straightforward to translate the string we use for "unknown", and we naturally would.